prow/github/report/report.go

```diff
@@ -282,19 +311,21 @@ func createEntry(pj prowapi.ProwJob) string {
 // createComment take a ProwJob and a list of entries generated with
 // createEntry and returns a nicely formatted comment. It may fail if template
 // execution fails.
-func createComment(reportTemplate *template.Template, pj prowapi.ProwJob, entries []string) (string, error) {
+func createComment(reportTemplate *template.Template, pjs []prowapi.ProwJob, entries []string) (string, error) {
 	plural := ""
 	if len(entries) > 1 {
 		plural = "s"
 	}
 	var b bytes.Buffer
-	if reportTemplate != nil {
-		if err := reportTemplate.Execute(&b, &pj); err != nil {
+	// Templating was default to report a single prowjob, which doesn't apply when
+	// multiple prowjobs are expected to be reported.
+	if reportTemplate != nil && len(pjs) == 1 {
+		if err := reportTemplate.Execute(&b, &pjs[0]); err != nil {
```

Comment:
FWIW the report template is usually related to the PR not a specific PJ, even though it is using the PJ in the template. This is kind of unfortunate and doesn't really make sense given that we maintain one failure comment on PRs, not one per PJ.
So we might be better off using the first PJ and still executing the template even if there are multiple PJs. I'm not sure what is better here so I'll leave this to your discretion. Either way this is something we can change later if needed. 
Here is an example reportTemplate for reference:
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=test-infra&pr=23693). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3AchaodaiG). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/test-infra/issues?q=is:issue+is:open) when you hit one in your PR.

Reply:
good point, shamelessly copy/pasted your comment as comment in the code, will go ahead and use the first PR as args for the template. Unit tests will be added